IDR/production/1.compute_statistics.py

```diff
@@ -0,0 +1,102 @@
+import pandas as pd
+import pathlib, sys
+from tqdm import tqdm
+from utils.statistics import collect_databank_stats, collect_study_stats
+
+# Define path to extraction_utils directory
+parent_dir = str(pathlib.Path(__file__).parents[1])
+sys.path.append(parent_dir)
+from metadata_extraction.extraction_utils.io import walk
+
+
+def nested_dict_pairs_iterator(elements_and_counts_dict):
+    """Collects all combinations of tuples for a nested dictionary
+
+    Parameters
+    ----------
+    elements_and_counts_dict: dict
+        Nested dictionary of all combinations of all unique entries of a metadata file
+        E.g. {study_name: {screen_id: {image_attribute: {unique_element: count}}}}
+
+    Returns
+    -------
+    All possible combinations of tuples for a nested dictionary
+        E.g. (study_name1, screen_id1, image_attribute1, unique_element1, count1)
+    """
+    # Iterate over all key-value pairs of dict argument
+    for key, value in elements_and_counts_dict.items():
+        # Check if value is of dict type
+        if isinstance(value, dict):
+            # If value is dict then iterate over all its values
+            for pair in nested_dict_pairs_iterator(value):
+                yield (key, *pair)
+        else:
+            # If value is not dict type then yield the value
+            yield (key, value)
+
+
+if __name__ == "__main__":
+    # Define study metadata directory
+    studies_metadata_dir = pathlib.Path("IDR/data/metadata")
+
+    # Collect metadata file paths
+    metadata_files = list(walk(studies_metadata_dir))
+    metadata_files = metadata_files[0:5]
+
+    # Make directories
+    stats_dir = pathlib.Path("IDR/data/statistics")
+    pathlib.Path.mkdir(stats_dir, exist_ok=True)
+
+    # Calculate statistics for each image attribute for individual studies
+    individual_study_stats = list()
+    unique_elements_and_counts = dict()
+
+    # Iterate through each study/screen/well.json metadata file
+    for metadata_path in metadata_files:
+        study_name = ((str(metadata_path).split("/")[-1]).split(".")[0]).split("_")[0]
+        screen_id = ((str(metadata_path).split("/")[-1]).split(".")[0]).split("_")[-1]
```

Comment:
Remember [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

```suggestion
        parsed_metadata_path = ((str(metadata_path).split("/")[-1]).split(".")[0]).split("_")
        study_name = parsed_metadata_path[0]
        screen_id = parsed_metadata_path[-1]
```

Something like this avoids repetition

Reply:
I see. I changed this according to what you suggested above.

Added in 51932618be844526c7398ce98110dc911c8d4b02.